basic enhancements for import logging

kyracho opened the pull request:
## Summary
Hello, this is a follow up PR to #192234 . The previous PR added simplistic logging to the saved objects importer. The goal now is to enhance the logs with information on the saved objects being imported, how they are imported, and by displaying any errors.

#### `import_saved_objects.ts`:
- Logs specific types being imported
- Logs size limit and overwrite status
- Logs Success/Fail messages

#### Changes to `saved_objects_importer.ts`:
- Passes the logger to `importSavedObjectsFromStream` 
- Removes "starting import"

#### Changes to `import_saved_objects.test.ts`:
- Updates it for the new logger parameter

#### Changes to `import.test.ts`:
- Uses the mock logger provided by core, instead of using a custom one

### Checklist

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#_add_your_labels)
- [ ] This will appear in the **Release Notes** and follow the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)



<!--ONMERGE {"backportTargets":["8.x"]} ONMERGE-->

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!

TinaHeiligers commented:
@elasticmachine merge upstream

TinaHeiligers commented:
@kyracho Thanks for following up on this, I'm a bit swamped right now but I'll get to it, although probably only next week.

kyracho commented:
@TinaHeiligers No problem at all, take your time! I’ll look forward to your feedback whenever you have the time. Appreciate it!

TinaHeiligers commented on packages/core/saved-objects/core-saved-objects-import-export-server-internal/src/import/saved_objects_importer.ts:
```diff
@@ -62,7 +62,16 @@ export class SavedObjectsImporter implements ISavedObjectsImporter {
+      .map((type) => type.name);
+    // Log the allowed types, similar to how the exporter logs types
+    this.#log.debug(`Initiating import process.`);
+    this.#log.debug(`Allowed types: [${allowedTypes.join(', ')}]`);
```
move onto `importSavedObjectsFromStream` and only log the types being imported:
```
// in `packages/core/saved-objects/core-saved-objects-import-export-server-internal/src/import/import_saved_objects.ts
...
 // Get the objects to import
  const collectSavedObjectsResult = await collectSavedObjects({
    readStream,
    objectLimit,
    supportedTypes,
    managed,
  });
log.debug(`Importing types: ${collectSavedObjectsResult.collectedObjects.map((title) => title ?? '')}`);
```

TinaHeiligers commented on packages/core/saved-objects/core-saved-objects-import-export-server-internal/src/import/saved_objects_importer.ts:
```diff
@@ -62,7 +62,16 @@ export class SavedObjectsImporter implements ISavedObjectsImporter {
+    // Log the allowed types, similar to how the exporter logs types
+    this.#log.debug(`Initiating import process.`);
+    this.#log.debug(`Allowed types: [${allowedTypes.join(', ')}]`);
+    this.#log.info(`Import size limit: ${this.#importSizeLimit} saved objects.`);
```
don't log this

TinaHeiligers commented on packages/core/saved-objects/core-saved-objects-import-export-server-internal/src/import/saved_objects_importer.ts:
```diff
@@ -62,7 +62,16 @@ export class SavedObjectsImporter implements ISavedObjectsImporter {
+    this.#log.debug(`Allowed types: [${allowedTypes.join(', ')}]`);
+    this.#log.info(`Import size limit: ${this.#importSizeLimit} saved objects.`);
+    const overwriteStatus = overwrite ? 'enabled' : 'disabled';
+    this.#log.info(`Automatic overwrite is ${overwriteStatus} for the current import operation.`);
```
Combine the info logs into one:
```suggestion
    this.#log.info(`Importing with overwrite ${overwriteStatus} and size limit ${this.#importSizeLimit}`);
```

kyracho replied:
Thank you for your detailed feedback @TinaHeiligers, I really appreciate your patience.

TinaHeiligers commented on packages/core/saved-objects/core-saved-objects-import-export-server-internal/src/import/saved_objects_importer.ts:
```diff
@@ -75,7 +84,19 @@ export class SavedObjectsImporter implements ISavedObjectsImporter {
-    });
+    })
+      .then((response: SavedObjectsImportResponse) => {
+        this.#log.info(`🌸 Successfully imported ${response.successCount} saved objects.`);
```
Move the logging into `importSavedObjectsFromStream` and  only log plain text:
```suggestion
        this.#log.info(`Successfully imported ${response.successCount} saved objects.`);
```

TinaHeiligers commented on packages/core/saved-objects/core-saved-objects-import-export-server-internal/src/import/saved_objects_importer.ts:
```diff
@@ -75,7 +84,19 @@ export class SavedObjectsImporter implements ISavedObjectsImporter {
+        return response;
+      })
+      .catch((error) => {
+        this.#log.error('Failed to import saved objects');
```
Log the errors from within `importSavedObjectsFromStream` [before](https://github.com/elastic/kibana/blob/d94bf9a41b264fd0fd268d20317dc27f5678c2ae/packages/core/saved-objects/core-saved-objects-import-export-server-internal/src/import/import_saved_objects.ts#L187-L195) the `return`

TinaHeiligers reviewed (changes requested):
We need to combine logs with the same log level to reduce logging noise.
Logging needs to be within importSavedObjectsFromStream

TinaHeiligers commented on packages/core/saved-objects/core-saved-objects-import-export-server-internal/src/import/import_saved_objects.test.ts:
```diff
@@ -72,7 +72,17 @@ describe('#importSavedObjectsFromStream', () => {
   let typeRegistry: jest.Mocked<ISavedObjectTypeRegistry>;
   const namespace = 'some-namespace';
-
+  const mockLog = {
```
There is an existing mock for logging, prefer using that to a custom mock. 
```
import { loggerMock, type MockedLogger } from '@kbn/logging-mocks';
```

TAL at how it's done in https://github.com/elastic/kibana/blob/main/packages/core/saved-objects/core-saved-objects-import-export-server-internal/src/export/saved_objects_exporter.test.ts

TinaHeiligers reviewed (commented):
Core provides mocks

kyracho commented:
Thanks again for the detailed explanations, @TinaHeiligers . I really appreciate your patience. I addressed your suggestions. Just to confirm, we're not aiming to specifically test the logging behavior, right?

TinaHeiligers commented:
buildkite test this

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/246499)
* Commit: 80893e1592b77b154b430a96428d46606005aeda

### Failed CI Steps
* [Linting](https://buildkite.com/elastic/kibana-pull-request/builds/246499#0192d4e2-98d3-4115-850c-79eb729d8530)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/246499","number":246499,"commit":"80893e1592b77b154b430a96428d46606005aeda"}],"number":246499}
buildkite-pr-comment-->

kyracho commented:
Could someone with Buildkite access please share the details of the linting errors? Or, would it be possible for ElasticMachine to post the error details? Thank you!

afharo commented:
/ci

afharo commented:
/ci

TinaHeiligers commented:
buildkite test this

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/247144)
* Commit: 3113d2775035af929d0a533070597fa5733f7b52

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #246499](https://buildkite.com/elastic/kibana-pull-request/builds/246499) failed 80893e1592b77b154b430a96428d46606005aeda

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/247144","number":247144,"commit":"3113d2775035af929d0a533070597fa5733f7b52"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/246499","number":246499,"commit":"80893e1592b77b154b430a96428d46606005aeda"}],"number":247144}
buildkite-pr-comment-->

kyracho commented:
@TinaHeiligers @afharo TYSM!  ❤️

If things are ok for this PR, I'd like to choose another Kibana issue to work on.  That is, unless you have any suggestions for the next issue!

TinaHeiligers reviewed (approved):
LGTM on CI green. Logging enhancements are ongoing.

TinaHeiligers commented:
@elasticmachine merge upstream

TinaHeiligers commented:
buildkite test this

TinaHeiligers commented:
@kyracho this PR will merge automatically if CI is green.

kibanamachine commented:
Starting backport for target branches: 8.x

https://github.com/elastic/kibana/actions/runs/11622237294

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/247665)
* Commit: 70805ee7cc476367d49918ef1658ccdc81de4c5b

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :green_heart: [Build #247144](https://buildkite.com/elastic/kibana-pull-request/builds/247144) succeeded 3113d2775035af929d0a533070597fa5733f7b52
* :broken_heart: [Build #246499](https://buildkite.com/elastic/kibana-pull-request/builds/246499) failed 80893e1592b77b154b430a96428d46606005aeda

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/247665","number":247665,"commit":"70805ee7cc476367d49918ef1658ccdc81de4c5b"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/247144","number":247144,"commit":"3113d2775035af929d0a533070597fa5733f7b52"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/246499","number":246499,"commit":"80893e1592b77b154b430a96428d46606005aeda"}],"number":247665}
buildkite-pr-comment-->

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/198634">](https://github.com/elastic/kibana/pull/198634)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 9.4.3 -->
